language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -32,19 +36,84 @@ import           Language.PlutusCore.View
 import           PlutusPrelude
 
 import           Control.Lens.TH                                  (makeLenses)
+import Control.Lens.Setter
 import           Control.Monad.Except
-import           Control.Monad.Reader
 import qualified Data.Map                                         as Map
+import Control.Monad.RWS.Strict
+import Data.Tuple.Select
+import Data.HashMap.Monoidal
+import Data.Functor.Foldable
+import Language.PlutusCore.Evaluation.Machine.GenericSemigroup
+import Foreign.Storable
+import qualified Data.Text                             as T
+import qualified Data.ByteString.Lazy           as BSL
 
 type Plain f = f TyName Name ()
+type WithMemory f = f TyName Name ExMemory
+
+class ExMemoryUsage a where
```

Comment:
@michaelpj mentioned there's some code in the ledger that contains something similar

Reply:
https://github.com/input-output-hk/cardano-prelude/blob/master/src/Cardano/Prelude/HeapWords.hs#L64-L65